### DISCUSSION TITLE
Welcome to mopeka_pro_check Discussions!

### DISCUSSION THREAD

**Discussion by spbrogan** [spbrogan/mopeka_pro_check#1] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Muskrat775** [comment]
I'm glad to see that someone was able to get some help from Mopeka. I have contacted them twice with no luck. I have been wanting to integrate the Standard Tank Sensors into HA, but haven't had any luck as of yet. Does your sensor.mopeka_pro_check work with the standard sensors as well? If not, do you plan on creating a HA addon for the standard sensors?

**Reply by spbrogan** [reply]
Mopeka told me their sensors prior to the new Pro had inconsistent data formats.  That doesn't really make a lot of sense given that their app still had to decode it but to be honest I haven't tried.  I have an old sensor and was thinking of dumping the advertisement data and seeing how similar it is to the new sensor.  You could try to decode one of your sensors using the class defined in this library and see what happens.  I could easily see detecting the sensor version and changing the decoding logic. If you have changes and/or a PR feel free to submit it.

**Reply by mhaberler** [reply]
@spbrogan whom were you talking to at Mopeka and how? I am unable to entice ANY response from them after 10+ tries

nice product, but completely unresponsive outfit - my contractor wants a custom sensor and I think by now Mopeka is hopeless

**Comment by realhagie** [comment]
HI,
same question here - I like to add data of the standard sensor to my HA system. 
Seeing the adv. packet with a scanner but not be able to decode it. 
What I know right now is : 
raw packet : 0x1AFF0D000002A725457C50FE07EB6FA04106117C9040059D504FD00302A0AD
splits to 
Length 1A 
Type FF 
Data 0D00 0002A725457C50FE07EB6FA04106117C9040059D 504FD0 
Length 03 
Type 02 
Data A0AD
where 0D00 is the manufacturer and 504FD0 are the last 3 numbers of the mac address of the sensor
I have not tried to use the provided lib, I also contacted mopeka yesterday.

**Reply by realhagie** [reply]
Addition:
I was playing around with ble advertising from Rpi to proxy the information. This way I could "play" around with the values and see what happens in the App.

What I know right now is :
raw packet : 0x1AFF0D000002A725457C50FE07EB6FA04106117C9040059D504FD00302A0AD
splits to
Length 1A
Type FF
Data 0D00 0002 A7 25457C50FE 07EB6FA04106117C9040059D 504FD0
A7 = Battery Status
25457C50FE  These Bytes represent the fill height AND the sensor quality. I have not found out how they correlate. 

Length 03
Type 02
Data A0AD

where 0D00 is the manufacturer and 504FD0 are the last 3 numbers of the mac address of the sensor

**Comment by realhagie** [comment]
there is a new GIT Entry - this looks like the source code of the tank check app for android and ios :

https://github.com/carloswork1004/Mopeka_React-Native

maybe it's possible to RE the non-pro sensor data this way.
